Bottom: 3773213b5371853cc561f76a6c125ee7961adf4b
Top:    c8131841c4408d2abb5c97e5b8b2c21c82f21c38
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2013-08-22 11:56:27 -0700

Refresh of cmtime-to

---

diff --git a/examples/cmtime.c b/examples/cmtime.c
index fb6bb19..7342666 100644
--- a/examples/cmtime.c
+++ b/examples/cmtime.c
@@ -51,6 +51,7 @@ static struct rdma_event_channel *channel;
 static char *port = "7471";
 static char *dst_addr;
 static char *src_addr;
+static int timeout = 2000;
 
 enum step {
 	STEP_CREATE_ID,
@@ -374,7 +375,7 @@ static int run_client(void)
 	for (i = 0; i < connections; i++) {
 		start_perf(&nodes[i], STEP_RESOLVE_ADDR);
 		ret = rdma_resolve_addr(nodes[i].id, rai->ai_src_addr,
-					rai->ai_dst_addr, 2000);
+					rai->ai_dst_addr, timeout);
 		if (ret) {
 			perror("failure getting addr");
 			nodes[i].error = 1;
@@ -393,7 +394,7 @@ static int run_client(void)
 		if (nodes[i].error)
 			continue;
 		start_perf(&nodes[i], STEP_RESOLVE_ROUTE);
-		ret = rdma_resolve_route(nodes[i].id, 2000);
+		ret = rdma_resolve_route(nodes[i].id, timeout);
 		if (ret) {
 			perror("failure resolving route");
 			nodes[i].error = 1;
@@ -464,7 +465,7 @@ int main(int argc, char **argv)
 
 	hints.ai_port_space = RDMA_PS_TCP;
 	hints.ai_qp_type = IBV_QPT_RC;
-	while ((op = getopt(argc, argv, "s:b:c:p:")) != -1) {
+	while ((op = getopt(argc, argv, "s:b:c:p:t:")) != -1) {
 		switch (op) {
 		case 's':
 			dst_addr = optarg;
@@ -478,12 +479,16 @@ int main(int argc, char **argv)
 		case 'p':
 			port = optarg;
 			break;
+		case 't':
+			timeout = atoi(optarg);
+			break;
 		default:
 			printf("usage: %s\n", argv[0]);
 			printf("\t[-s server_address]\n");
 			printf("\t[-b bind_address]\n");
 			printf("\t[-c connections]\n");
 			printf("\t[-p port_number]\n");
+			printf("\t[-t timeout_ms]\n");
 			exit(1);
 		}
 	}
